### PR TITLE
[AC-7752]: Create Utility for obtaining the user id to be used for new React apps

### PR DESCRIPTION
**Pull request by PrinceDavis**
https://masschallenge.atlassian.net/browse/AC-7752

Create Utility for obtaining the user id to be used for new React apps

Testing notes: 
- make a graphQL request for either an expert or an entreprenuer without providing an id and see that server does not break
- Fire up directory and see that graphQL server still response to queries with ID

### CODE REVIEW

**Review by shakirandagire — APPROVED** [review]
The code changes look good to me

File: web/impact/impact/graphql/query.py [hunk 1]
```diff
@@ -34,7 +35,7 @@ def resolve_expert_profile(self, info, **kwargs):
         return expert
 
     def resolve_entrepreneur_profile(self, info, **kwargs):
-        user_id = kwargs.get('id')
+        user_id = kwargs.get('id') or info.context.user.id
```
Comment (jonkiparsky) [inline web/impact/impact/graphql/query.py]: use the default option for kwargs.get:

`user_id = kwargs.get('id', info.context.user.id)`

File: web/impact/impact/graphql/query.py [hunk 2]
```diff
@@ -25,7 +25,8 @@ class Query(graphene.ObjectType):
         EntrepreneurProfileType, id=graphene.Int())
 
     def resolve_expert_profile(self, info, **kwargs):
-        user_id = kwargs.get('id')
+        user_id = kwargs.get('id') or info.context.user.id
```
Comment (jonkiparsky) [inline web/impact/impact/graphql/query.py]: use the default option for kwargs.get:

`user_id = kwargs.get('id', info.context.user.id)`

File: web/impact/impact/tests/test_graphql.py [hunk 3]
```diff
@@ -658,6 +658,60 @@ def test_current_judge_cannot_view_mentor_profile(self):
...
+        user_role = get_user_role_by_name(UserRole.FINALIST)
+        program_role = ProgramRoleFactory.create(
+            user_role=user_role,
+            program__program_status=ACTIVE_PROGRAM_STATUS
+        )
+        ProgramRoleGrantFactory(person=user,
+                                program_role=program_role)
+        user.set_password('password')
+        user.save()
+        query = """
```
Comment (jonkiparsky) [inline web/impact/impact/tests/test_graphql.py]: would be nice to DRY this out, see comment above.

File: web/impact/impact/tests/test_graphql.py [hunk 4]
```diff
@@ -658,6 +658,60 @@ def test_current_judge_cannot_view_mentor_profile(self):
         self._assert_expert_cannot_view_profile(UserRole.JUDGE,
                                                 UserRole.MENTOR)
 
+    def test_loggedin_expert_data_is_returned_on_missing_id(self):
+        user = expert_user(UserRole.MENTOR)
+        query = """
```
Comment (jonkiparsky) [inline web/impact/impact/tests/test_graphql.py]: Can we DRY out this repeated code?
The query and expected_json defined here are pretty stereotyped. `query` could probably be a constant and `expected_json` could be a function taking the user and possibly the key string, in this case "expertProfile" and returning a dict of this shape.

File: web/impact/impact/tests/test_graphql.py [hunk 5]
```diff
@@ -658,6 +658,60 @@ def test_current_judge_cannot_view_mentor_profile(self):
...
+                'expertProfile': {
+                    'user': {
+                        'lastName': user.last_name
+                    }
+                }
+            }
+        }
+        self._assert_response_equals_json(query, expected_json, email=user.email)
+
+    def test_loggedin_entrepnur_data_is_returned_on_missing_id(self):
```
Comment (jonkiparsky) [inline web/impact/impact/tests/test_graphql.py]: s/b test_logged_in_entrepreneur_data_is_returned_on_missing_id

**Review by jonkiparsky — CHANGES_REQUESTED** [review]
Change looks good. Would be nice to clean up the tests a bit.

File: web/impact/impact/tests/test_graphql.py (hunk 5 above)
Comment (PrinceDavis) [reply web/impact/impact/tests/test_graphql.py]: Say more here @jonkiparsky what is s/b?

Comment (jonkiparsky) [reply web/impact/impact/tests/test_graphql.py]: sorry, "s/b" stands for "should be"

**Review by jonkiparsky — APPROVED** [review]

**Pull request merged**